src/foundry/foundry.js/collections/documentCollections/compendiumCollection.d.ts

```diff
@@ -0,0 +1,216 @@
+     * @param options - Additional options which modify the Compendium creation request
+     *                  default `{}`
+     */
+    static createCompendium<T extends Metadata<any>>(
+      metadata: T,
+      options?: Partial<DocumentModificationOptions>
+    ): Promise<CompendiumCollection<T>>;
+
+    /**
+     * Assign configuration metadata settings to the compendium pack
+     * @param settings - The object of compendium settings to define
+     *                   default: `{}`
+     * @returns A Promise which resolves once the setting is updated
+     */
+    configure(settings?: Partial<CompendiumConfiguration>): Promise<void>;
+
+    /**
+     * Delete an existing world-level Compendium Collection.
+     * This action may only be performed for world-level packs by a Gamemaster User.
+     */
+    deleteCompendium(): Promise<this>;
+
+    /**
+     * Duplicate a compendium pack to the current World.
+     * @param label - A new Compendium label
+     */
+    duplicateCompendium({ label }?: { label?: string }): Promise<this>;
+
+    /**
+     * Validate that the current user is able to modify content of this Compendium pack
+     * @param requireUnlocked - `(default: true)`
+     */
+    protected _assertUserCanModify({ requireUnlocked }?: { requireUnlocked?: boolean }): boolean;
+
+    /** Request that a Compendium pack be migrated to the latest System data template */
+    migrate(options?: object): Promise<this>;
+
+    _onCreateDocuments(
+      documents: DocumentInstanceForMetadata<T>[],
+      result: DocumentInstanceForMetadata<T>['data']['_source'][],
+      options: DocumentModificationOptions,
+      userId: string
+    ): void;
+
+    _onUpdateDocuments(
+      documents: DocumentInstanceForMetadata<T>[],
+      result: DeepPartial<DocumentInstanceForMetadata<T>>[],
+      options: DocumentModificationOptions,
+      userId: string
+    ): void;
+
+    _onDeleteDocuments(
+      documents: DocumentInstanceForMetadata<T>[],
+      result: string[],
+      options: DocumentModificationOptions,
+      userId: string
+    ): void;
+
+    /**
+     * Follow-up actions taken when Documents within this Compendium pack are modified
+     */
+    protected _onModifyContents(
+      documents: DocumentInstanceForMetadata<T>[],
+      options: DocumentModificationOptions,
+      userId: string
+    ): void;
+
+    /** @deprecated since 0.8.0 */
+    get entity(): this['documentClass']['documentName'];
+
+    /** @deprecated since 0.8.0 */
+    get content(): ReturnType<this['getDocuments']>;
+
+    /** @deprecated since 0.8.0 */
+    getEntry(id: string): Promise<DocumentInstanceForMetadata<T>['data']>;
+
+    /** @deprecated since 0.8.0 */
+    getEntity(id: string): ReturnType<this['getDocument']>;
+
+    /** @deprecated since 0.8.0 */
+    importEntity(document: DocumentInstanceForMetadata<T>): ReturnType<this['importDocument']>;
+
+    /** @deprecated since 0.8.0 */
+    createEntity(
+      data: any,
```

Comment:
it should be the constructor type for the configured document instance. You can get it hacky trick:
```ts
    createEntity(
      data: Parameters<DocumentInstanceForMetadata<T>['data']['_initializeSource']>[0],
      options?: Partial<DocumentModificationOptions>
    ): ReturnType<this['documentClass']['create']>;
```
We probably should create a helper type for that as this is not very obvious... :D